src/algebra/order/to_interval_mod.lean

```diff
@@ -733,3 +733,166 @@ lemma to_Ico_mod_zero_one (x : α) : to_Ico_mod (0 : α) zero_lt_one x = int.fra
+by simpa only [zero_add] using Union_Ico_add_zsmul 0 hb
+
+lemma Union_Icc_add_zsmul [archimedean α] (a : α) {b : α} (hb : 0 < b) :
+  (⋃ (n : ℤ), Icc (a + n • b) (a + (n + 1) • b)) = univ :=
+by simpa only [Union_Ioc_add_zsmul a hb, univ_subset_iff] using
+  Union_mono (λ n : ℤ, (Ioc_subset_Icc_self : Ioc (a + n • b) (a + (n + 1) • b) ⊆ Icc _ _))
+
+lemma Union_Icc_zsmul [archimedean α] {b : α} (hb : 0 < b) :
+  (⋃ (n : ℤ), Icc (n • b) ((n + 1) • b)) = univ :=
+by simpa only [zero_add] using Union_Icc_add_zsmul 0 hb
+
+lemma pairwise_disjoint_Ioc_add_zsmul (a b : α) :
+  pairwise (disjoint on (λ (n : ℤ), Ioc (a + n • b) (a + (n + 1) • b))) :=
+begin
+  -- The assumption `linear_ordered_add_comm_group α` can be weakened to `ordered_add_comm_group α`
+  -- if we assume that `0 < b`, but `Ioc` is not a very useful concept in partial orders anyway.
+  simp_rw [function.on_fun, set.disjoint_iff],
+  intros m n hmn x hx,
+  apply hmn,
+  rcases lt_or_le 0 b with hb | hb,
+  { have i1 := hx.1.1.trans_le hx.2.2,
+    have i2 := hx.2.1.trans_le hx.1.2,
+    rw [add_lt_add_iff_left, zsmul_lt_zsmul_iff hb, int.lt_add_one_iff] at i1 i2,
+    exact le_antisymm i1 i2 },
+  { -- case b ≤ 0 : vacuous but true
+    have : ∀ (n : ℤ), Ioc (a + n • b) (a + (n + 1) • b) = ∅,
+    { intro n,
+      apply Ioc_eq_empty_of_le,
+      rw [add_zsmul, one_zsmul, add_le_add_iff_left],
+      exact add_le_of_nonpos_right hb },
+    simp_rw [this, empty_inter] at hx,
+    exact hx.elim }
+end
+
+lemma pairwise_disjoint_Ioc_zsmul (b : α) :
+  pairwise (disjoint on (λ (n : ℤ), Ioc (n • b) ((n + 1) • b))) :=
+by simpa only [zero_add] using pairwise_disjoint_Ioc_add_zsmul 0 b
+
+lemma pairwise_disjoint_Ico_add_zsmul (a b : α) :
+  pairwise (disjoint on (λ (n : ℤ), Ico (a + n • b) (a + (n + 1) • b))) :=
+begin
+  simp_rw [function.on_fun, set.disjoint_iff],
+  intros m n hmn x hx,
+  apply hmn,
+  rcases lt_or_le 0 b with hb | hb,
+  { have i1 := hx.1.1.trans_lt hx.2.2,
+    have i2 := hx.2.1.trans_lt hx.1.2,
+    rw [add_lt_add_iff_left, zsmul_lt_zsmul_iff hb, int.lt_add_one_iff] at i1 i2,
+    exact le_antisymm i1 i2 },
+  { have : ∀ (n : ℤ), Ico (a + n • b) (a + (n + 1) • b) = ∅,
+    { intro n,
+      apply Ico_eq_empty_of_le,
+      rw [add_zsmul, one_zsmul, add_le_add_iff_left],
+      exact add_le_of_nonpos_right hb },
+    simp_rw [this, empty_inter] at hx,
+    exact hx.elim }
+end
+
+lemma pairwise_disjoint_Ico_zsmul (b : α) :
+  pairwise (disjoint on (λ (n : ℤ), Ico (n • b) ((n + 1) • b))) :=
+by simpa only [zero_add] using pairwise_disjoint_Ico_add_zsmul 0 b
+
+lemma pairwise_disjoint_Ioo_add_zsmul (a b : α) :
+  pairwise (disjoint on (λ (n : ℤ), Ioo (a + n • b) (a + (n + 1) • b))) :=
+λ m n hmn, (pairwise_disjoint_Ioc_add_zsmul a b hmn).mono Ioo_subset_Ioc_self Ioo_subset_Ioc_self
+
+lemma pairwise_disjoint_Ioo_zsmul (b : α) :
+  pairwise (disjoint on (λ (n : ℤ), Ioo (n • b) ((n + 1) • b))) :=
+by simpa only [zero_add] using pairwise_disjoint_Ioo_add_zsmul 0 b
```

Comment:
Those lemmas (and the ones in the `linear_ordered_ring` section that don't need `archimedean`) belong in an earlier file. Look around `data.set.intervals`.

Reply:
Maybe `data.set.intervals.disjoint` would be a possibility, or `algebra.group_power.lemmas`. The problem is that these files, and basically all the other low-level files about orders, are frozen for the mathlib4 port. What do you suggest?

Reply:
They're not really frozen; you just have to forward-port the results. For this PR which would just adds a big block of lemmas, that's a really easy task. If your PR moves lemmas around then it's a lot worse.

Reply:
> that's a really easy task

Yeah, right, so learning an entire new language -- and setting up all its toolchain on my computer without simultaneously breaking the old one -- is trivial. A five-minute job. 

The Greek gods condemned Sisyphus to endlessly push a boulder up a hill; but if they had really wanted to torture him, they should have given him a nontrivial mathematical proof and told him to PR it to mathlib.

Reply:
> Yeah, right, so learning an entire new language -- and setting up all its toolchain on my computer without simultaneously breaking the old one -- is trivial. A five-minute job.

Sorry, it sounds like I've sent you down a painful path by accident. You don't need to learn a new language or install anything on your computer. Instead, you can:

* Write your Lean3 code as normal and get this PR merged
* Make a blank placeholder Lean4 PR
* Wait for `mathlib3port` to autoport your code to Lean3 for you
* Open the mathlib4 repo in gitpod (so that you don't need to install Lean4)
* Copy the block of code from mathlib3port into the right place in the file, and fix any errors

Reply:
The error is that `pairwise` should be `Pairwise`; but `mathlib3port` would have made that correction for you.

Reply:
OK, thanks for the advice! I've moved the lemmas `data.set.intervals.group`, and made a matching mathlib4 PR at https://github.com/leanprover-community/mathlib4/pull/2263.